### PR TITLE
Expose settle with beneficiary transaction info

### PR DESCRIPTION
**Pull request by mdomasevicius**
- a way to show settle with beneficiary status NodeUI
- make settle with beneficiary async since most users reached timeouts

### CODE REVIEW

File: core/beneficiary/saver.go [hunk 1]
```diff
@@ -61,23 +63,77 @@ type l1Saver struct {
...
 	}
 }
 
+// SettleWithBeneficiaryState represents the state of settle with beneficiary transaction
+type SettleWithBeneficiaryState string
+
+const bucketBeneficiaryChangeStatus = "beneficiary-change-status"
+const (
+	// PENDING transaction is pending
+	PENDING SettleWithBeneficiaryState = "pending"
```
Comment (tomasmik) [inline core/beneficiary/saver.go:84]: Standard library uses camel-case for constants or any other variable except maybe for some vars in `os` package, so I would suggest to keep it like that.

File: core/beneficiary/saver.go [hunk 2]
```diff
@@ -61,23 +63,77 @@ type l1Saver struct {
...
+	var errorMSG string
+	if settleError != nil {
+		errorMSG = settleError.Error()
+	}
+	err := bcs.st.SetValue(bucketBeneficiaryChangeStatus, storageKey(bcs.chainID, id.Address), &BeneficiaryChangeStatus{
+		State: sbs,
+		Error: errorMSG,
+	})
+
+	log.Err(err).Msg("Failed to update BeneficiaryChangeStatus")
```
Comment (tomasmik) [inline core/beneficiary/saver.go:110]: Forgot an `if` here?

File: core/beneficiary/saver.go [hunk 3]
```diff
@@ -61,23 +63,77 @@ type l1Saver struct {
...
+// BeneficiaryChangeStatus holds Beneficiary settlement transaction information
+type BeneficiaryChangeStatus struct {
+	State SettleWithBeneficiaryState
+	Error string
+}
+
+func (bcs *beneficiaryChangeKeeper) updateBeneficiaryChangeStatus(id identity.Identity, sbs SettleWithBeneficiaryState, settleError error) {
+	var errorMSG string
+	if settleError != nil {
+		errorMSG = settleError.Error()
```
Comment (tomasmik) [inline core/beneficiary/saver.go:103]: Im gonna be annoying here and say that it should be `errorMsg` if we want to be super idiomatic about things 🤷🏻

File: core/beneficiary/saver.go [hunk 4]
```diff
@@ -61,23 +63,77 @@ type l1Saver struct {
...
+	err := bcs.st.SetValue(bucketBeneficiaryChangeStatus, storageKey(bcs.chainID, id.Address), &BeneficiaryChangeStatus{
+		State: sbs,
+		Error: errorMSG,
+	})
+
+	log.Err(err).Msg("Failed to update BeneficiaryChangeStatus")
+}
+
+// BeneficiaryChangeStatus get beneficiary change status for given identity
+func (bcs *beneficiaryChangeKeeper) BeneficiaryChangeStatus(id identity.Identity) *BeneficiaryChangeStatus {
```
Comment (tomasmik) [inline core/beneficiary/saver.go:114]: There are few things I would consider:
* Add `Get` or something like that in front of the function name to indicate what exactly it's doing
* Dont return possible `nil` values. Either return `(*BeneficiaryChangeStatus, bool)` or `(*BeneficiaryChangeStatus, error)` and possibly drop loging from here. From general practice, returning `nil` and no error results in unwanted panics :(

File: core/beneficiary/saver.go [hunk 5]
```diff
@@ -61,23 +63,77 @@ type l1Saver struct {
...
 	}
 
+	b.updateBeneficiaryChangeStatus(id, PENDING, nil)
+	settleError := b.set.SettleWithBeneficiary(b.chainID, id, beneficiary, hermesID)
+	b.updateBeneficiaryChangeStatus(id, COMPLETED, settleError)
+	if settleError != nil {
+		return settleError
+	}
+
 	return b.set.SettleWithBeneficiary(b.chainID, id, beneficiary, hermesID)
```
Comment (tomasmik) [inline core/beneficiary/saver.go:137]: forgot to remove this?

File: core/beneficiary/saver.go [hunk 6]
```diff
@@ -61,23 +63,77 @@ type l1Saver struct {
...
-		ad:      ad,
+		chainID:                 chainID,
+		set:                     set,
+		ad:                      ad,
+		beneficiaryChangeKeeper: beneficiaryChangeKeeper{st: st, chainID: chainID},
 	}
 }
 
+// SettleWithBeneficiaryState represents the state of settle with beneficiary transaction
+type SettleWithBeneficiaryState string
```
Comment (tomasmik) [inline core/beneficiary/saver.go:79]: gonna nag a little: the word `beneficiary` is already in the package name, so when you call this, you'll now call it like `beneficiary.SettleWithBeneficiaryState` I guess something like `beneficiary.SettlementState` would make more sense?

**Review by tomasmik — APPROVED** [review]

**Review by vkuznecovas — APPROVED** [review]

**Review by tadaskay — APPROVED** [review]

**Pull request merged**